### PR TITLE
feature bv-71/bar chart

### PR DESCRIPTION
**Pull request by dariuszsobkowicz**

### CODE REVIEW

**Review by wojciech-nowakowski — APPROVED** [review]

File: src/components/BarChart/BarChart-container.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      'BarChartSegmentColorPrivate', 
+      'BarChartSegmentColorDaap', 
+      'BarChartSegmentColorFraud'
+    ];
+
+    if (index < 10 && index === activeSegment) {
+      adjustedClasses.push(classes.BarChartSegmentActive);
+    }
+
+    if (!(index === activeSegment) && index === 0) {
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart-container.tsx]: Why not `index !==activeSegment`?

Comment (dariuszsobkowicz) [reply src/components/BarChart/BarChart-container.tsx]: good point 👍

File: src/components/BarChart/BarChart-container.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      adjustedClasses.push(classes.BarChartSegmentFirstInactive)
+    }
+
+    adjustedClasses.push(classes[colors[Math.round(Math.random() * 3)]])
+
+    return adjustedClasses.join(' ');
+  };
+
+  const data = wallets.reduce((acc: any, object: any, index: any) => {
+    if (!(object.percentage < 0.1)) {
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart-container.tsx]: `object.percentage >= 0.1` ?

File: src/components/BarChart/BarChart.module.css [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  background: linear-gradient(90deg, #5975FF 0%, #569AF5 100%);
+}
+
+.BarChartSegmentColorDaap {
+  background: linear-gradient(90deg, #74798C 0%, #A9B0CC 100%);
+}
+
+.BarChartSegmentColorFraud {
+  background: linear-gradient(90deg, #2A2D3A 0%, #34363F 100%);
+}
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart.module.css]: Use JSS instead of css.

File: src/theme/constants.ts [hunk 4]
```diff
@@ -7,3 +7,5 @@ export const DEFAULT_BG_COLOR = '#161B32';
 
 export const PRIMARY_MAIN_COLOR = '#252D48';
 export const SECONDARY_MAIN_COLOR = '#A3A7B0';
+
+export const BAR_CHART_WIDTH = 2000;
```
Comment (damianlewandowski) [inline src/theme/constants.ts]: Is this ever going to be used outside of BarChart component? If not, a better idea would be to put this inside that module to put it as close to where it's used as possible and reduce polution.

File: src/components/BarChart/BarChart-container.tsx (hunk 2 above)
Comment (dariuszsobkowicz) [reply src/components/BarChart/BarChart-container.tsx]: to reduce the number of others
others

File: src/components/BarChart/BarChart-container.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+  const getStyle: any = (acc: any, object: any) => ({
+    position: 'absolute',
+    height: '100%',
+    top: 0,
+    left: acc.position,
+    width: BAR_CHART_WIDTH * object.percentage / 100,
+  })
+
+  const getClasses = (index: any): any => {
```
Comment (damianlewandowski) [inline src/components/BarChart/BarChart-container.tsx]: use something like `clsx` instead

File: src/components/BarChart/BarChart-container.tsx (hunk 2 above)
Comment (damianlewandowski) [reply src/components/BarChart/BarChart-container.tsx]: I mean that it can be written as `object.percentage >= 0.1` instead of with additional negation

**Review by damianlewandowski — CHANGES_REQUESTED** [review]
More than half of the defined types are of type `any`. Is this a good idea to push it into the development? Most of them can be deduced at this point.

**Review by wojciech-nowakowski — APPROVED** [review]

**Review by damianlewandowski — APPROVED** [review]
Nice 💯

**Pull request merged**